aztec/src/main/kotlin/org/wordpress/aztec/watchers/DeleteMediaElementWatcher.kt

```diff
@@ -8,17 +8,38 @@ import java.lang.ref.WeakReference
 
 class DeleteMediaElementWatcher(aztecText: AztecText) : TextWatcher {
     private val aztecTextRef: WeakReference<AztecText?> = WeakReference(aztecText)
+    private var deleted = false
+    private var queueHasBeenPopulatedInThisTimeframe = false
+    private var deletedSpans = ArrayList<AztecMediaSpan>()
 
     override fun beforeTextChanged(text: CharSequence, start: Int, count: Int, after: Int) {
         if (aztecTextRef.get()?.isTextChangedListenerDisabled() ?: true) {
             return
         }
 
         if (count > 0) {
+            deleted = true
+
+            // we need to save the spans for later reference, as these spans are going to not exist anymore by the time
+            // the afterTextChanged event arrives. So, adding them to a local deletedSpans array.
             aztecTextRef.get()?.text?.getSpans(start, start + count, AztecMediaSpan::class.java)
                     ?.forEach {
-                        it.onMediaDeleted()
+                        deletedSpans.add(it)
+                    }
+
+            // only call the onMediaDeleted callback if we are sure the ObservationQueue has not been filled with
+            // platform-only events in a short time. These platform-originated events shall not be confused with
+            // real user deletions.
+            aztecTextRef.get()?.postDelayed( object : Runnable {
```

Comment:
This PR works fine, tested on API26+ and lower APIs, and I think we can merge it.
I have one consideration though. May I suggest to split `DeleteMediaElementWatcher.kt` in 2 classes, one for API26+ that does use `postDelayed` and the other for API25 and lower?

If you think it doesn't worth the effort I'm ok with this, but the Runnable is being executed a lot of times on API25 devices, even when no media are deleted, with `deletedSpans` empty, and this is another point that could slow down Aztec even more.